src/locutus/model/terminology.py

```diff
@@ -437,12 +452,15 @@ def add_provenance(
     ):
         if target is None:
             target = "self"
+            code_index = "self"
+        else:
+            code_index = get_code_index(target)

```

Comment:
Just being silly picky, get_code_index("self") would still be "self", right? I was told a long time ago that else is relatively expensive if it isn't necessary. So, you might be able to get away with just keeping 457 even if your code did exercise line 454. (of course, that would have to be shifted over)